pkg/segment/segexecution.go

```diff
@@ -54,6 +54,9 @@ func ExecuteMetricsQuery(mQuery *structs.MetricsQuery, timeRange *dtu.MetricsTim
 func ExecuteMultipleMetricsQuery(hashList []uint64, mQueries []*structs.MetricsQuery, queryOps []structs.QueryArithmetic, timeRange *dtu.MetricsTimeRange, qid uint64) *mresults.MetricsResult {
 	resMap := make(map[uint64]*mresults.MetricsResult)
 	for index, mQuery := range mQueries {
+		if _, ok := resMap[hashList[index]]; ok {
+			continue
+		}
```

Comment:
Is `ok == true` an normal state? If not, we probably want to log a warning

Reply:
Yes it is, there might be cases where the same query is executed multiple times. It is just to prevent that duplicate execution.